php/libraries/OutputWrapper.class.inc

```diff
@@ -0,0 +1,188 @@
+<?php declare(strict_types=1);
+/**
+ * This class is designed to assist in writing command-line interface tools
+ * for LORIS.
+ *
+ * It contains helper functions to format output, etc.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class OutputWrapper
+{
+    // Colours mapped to ANSI escape codes.
+    // @see https://en.wikipedia.org/wiki/ANSI_escape_code#3/4_bit
+    const COLOURS = [
+        'red'    => '31',
+        'yellow' => '33',
+        'green'  => '32',
+    ];
+    /**
+     * Whether this script should log its output to a file or just print to
+     * STDOUT.
+     *
+     * @var bool
+     */
+    private $_loggingEnabled = false;
+    /**
+     * The directory that will contain log files written by this script.
+     *
+     * @var SPLFileInfo
+     */
+    private $_logDirectory;
+
+    /**
+     * Create a new command line helper.
+     *
+     * @param ?SPLFileInfo $logDirectory The directory that will contain log
+     *                                   files written by this script.
+     *
+     * @return void
+     */
+    public function __construct(SPLFileInfo $logDirectory = null)
+    {
+        $this->_logDirectory = new SPLFileInfo(
+            $logDirectory ?? __DIR__ . '/tools/'
+        );
+    }
+
+    /**
+     * Print a formatted and indented bullet point.
+     *
+     * @param string $line The output.
+     *
+     * @return void
+     */
+    public function printBulletPoint(string $line): void
+    {
+        echo "\t* $line\n";
+    }
+
+    /**
+     * Prints a line of text with a newline before. Can optionally be printed
+     * to the command-line in coloured text. Valid options for colours are given
+     * in the COLOURS class constant.
+     * If logging is enabled, the line will also be written to a log file. Log
+     * files will not contain coloured text.
+     *
+     * @param string   $line   To be written.
+     * @param string   $colour A valid colour, given by COLOURS, for the output.
+     * @param resource $handle A file resource where text should be written.
+     *
+     * @return void
+     */
+    public function printLine(
+        string $line,
+        string $colour = '',
+        $handle = STDOUT
+    ) {
+        $output = $line;
+
+        // Verify that if a colour is supplied, it is valid
+        if (!empty($colour)
+            && !array_key_exists(strtolower($colour), self::COLOURS)
+        ) {
+            throw new \InvalidArgumentException(
+                "Invalid argument for colour (`$colour`)"
+            );
+        }
+
+        // If a valid colour is supplied and the output is going to a TTY,
+        // colourize the output.
+        if (!empty($colour) && stream_isatty($handle)) {
+            // Format the output to colour the whole line.
+            $output = sprintf("\n\e[%sm$line\e[0m", self::COLOURS[$colour]);
```

Comment:
Why is colour interpolated but $line is directly inserted in the formatting string? $line should be using a formatting verb too so that it doesn't get interpolated by sprintf.